auto_backup/models/db_backup.py

```diff
@@ -163,14 +163,15 @@ def action_backup(self):
         # Ensure a local backup exists if we are going to write it remotely
         sftp = self.filtered(lambda r: r.method == "sftp")
         if sftp:
-            if backup:
```

Comment:
It just seemed reasonable to do the backup once, cache it and send it in all configured locations; why are you now creating the backup again and again for each `sftp` configuration?

Reply:
I backported from [https://github.com/OCA/server-tools/commit/52b070e839fd83b60bd1ff4eef797d29bdcb02e7#diff-75853cf520b8f3ccf275c9a73a465d1fddc919bb908fa19627211491deb69f37](https://github.com/OCA/server-tools/commit/52b070e839fd83b60bd1ff4eef797d29bdcb02e7#diff-75853cf520b8f3ccf275c9a73a465d1fddc919bb908fa19627211491deb69f37), I changed from this version, I suppose may be changed for further versions. Correct?

Reply:
I restored the condition, replacing the _with_ statement into the loop, to allow file closing each step.